### PR TITLE
[RFC] `match_only_text` type migration - Stage 0

### PR DESCRIPTION
**Pull request by ebeahan**
### Summary

Stage 0 proposal for migrating `text` type fields, including the `message` base field, to use the `match_only_text` data type being introduced in Elasticsearch 7.14.

Stage 0 (Strawperson) Criteria:

- [x] Discuss with domain or subject matter experts the utility of these changes
- [x] Discuss with the ECS team whether these changes seem appropriate for ECS

[Preview of markdown proposal](https://github.com/ebeahan/ecs/blob/rfc/match_only_text-migrate/rfcs/text/0000-match_only_text-data-type.md)

### CODE REVIEW

File: rfcs/text/0000-match_only_text-data-type.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+Indexing `message` fields as the `text` type in security and application logs consumes significant disk space. Part of the disk space spent is on indexing to support phrase queries, which aren't often used in logging use cases. Elasticsearch 7.14 introduces a new field type called `match_only_text` which is a more space-efficient variant of the `text` field type for this logging-focused use cases.
```
Comment (jpountz) [inline rfcs/text/0000-match_only_text-data-type.md:16]:
```suggestion
Indexing `message` fields as the `text` type in security and application logs consumes significant disk space. Part of the disk space spent is on indexing to support scoring and phrase queries, which aren't often used in logging use cases. Elasticsearch 7.14 introduces a new field type called `match_only_text` which is a more space-efficient variant of the `text` field type for this logging-focused use cases.
```

File: rfcs/text/0000-match_only_text-data-type.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+* `index_options: docs`
+* `norms: false`
+
+`match_only_text` uses the `_source` for positional queries like `match_phrase`
+
+The `match_only_text` type supports the same feature set as `text`, except the following:
+
+* No support for scoring: queries ignore index statistics and produce constant scores.
+* Span queries are unsupported. If a span query is run, then a field mapped as match_only_text will be ignored from the search response.
```
Comment (jpountz) [inline rfcs/text/0000-match_only_text-data-type.md:52]:
```suggestion
* Span queries are unsupported. If a span query is run, then shards where the field is mapped as match_only_text will be returned as failed in the search response and their hits will be ignored.
```

**Review by jpountz — APPROVED** [review]
This looks great. I left a few minor clarifications.

File: rfcs/text/0023-match_only_text-data-type.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+`match_only_text` uses the `_source` for positional queries like `match_phrase`
+
+The `match_only_text` type supports the same feature set as `text`, except the following:
+
+* No support for scoring: queries ignore index statistics and produce constant scores.
+* Span queries are unsupported. If a span query is run, then shards where the field is mapped as match_only_text will be returned as failed in the search response and their hits will be ignored.
+* Phrase and intervals queries run slower.
+
+Like `text`, `match_only_text` fields do not support aggregations.
```
[the remark this answers is not in the document]
Comment (djptek) [reply rfcs/text/0023-match_only_text-data-type.md:55]: see related PR to Elasticsearch, I may cancel this review depending on the response from there also: https://github.com/elastic/elasticsearch/pull/72985

[the remark this answers is not in the document]
Comment (ebeahan) [reply rfcs/text/0023-match_only_text-data-type.md:55]: Great catch! I'll note this and include the change in the stage 1 PR for this RFC.

**Pull request merged**